src/anomaly/automl_regressors.rs

```diff
@@ -0,0 +1,161 @@
+        num_features: 6,
+        feature_names,
+        target_names,
+        description,
+        data: data.features.into_raw_vec(),
+        target: data.targets.into_raw_vec(),
+    }
+}
+
+#[derive(Default)]
+/// A metaml wrapper for an [automl] linear regressor
+pub struct LinearRegressor {
+    model: Option<SupervisedModel>,
+}
+
+impl LinearRegressor {
+    /// Constructs a new LinearRegressor
+    ///
+    /// # Examples
+    /// ```no_run
+    /// # use clam::anomaly::automl_regressors::LinearRegressor;
+    /// #
+    /// // Construct a new LinearRegressor
+    /// let regressor = LinearRegressor::new();
+    ///
+    /// // Train and use your regressor
+    /// todo!();
+    /// ```
+    pub fn new() -> Self {
+        Self { model: None }
+    }
+}
+
+impl MetaMLModel for LinearRegressor {
+    fn train(&mut self, data: MetaMLDataset) {
+        // Construct a metaml dataset from the input data
+        let dataset = metaml_dataset_to_automl(data);
+
+        // Create the settings and model
+        let settings = automl::Settings::default_regression().only(Algorithm::Linear);
+        let mut model = SupervisedModel::new(dataset, settings);
+
+        // Train the model
+        model.train();
+
+        // Store the model
+        self.model = Some(model);
+    }
+
+    fn predict(&self, features: [f32; 6]) -> f32 {
+        let model = self
+            .model
+            .as_ref()
+            .expect("Model must be trained before making predictions");
+
+        model.predict(vec![features.to_vec()])[0]
+    }
+
+    fn load(path: &str) -> Self {
+        let model = SupervisedModel::new_from_file(path);
+        LinearRegressor { model: Some(model) }
+    }
+
+    fn save(&self, path: &str) {
+        let model = self.model.as_ref().expect("Model must be trained before being saved.");
+        model.save(path);
+    }
+}
+
+#[derive(Default)]
+/// A metaml wrapper for an [automl] decision tree regressor
+pub struct DecisionTreeRegressor {
+    model: Option<SupervisedModel>,
+}
+
+impl DecisionTreeRegressor {
+    /// Constructs a new DecisionTreeRegressor
+    ///
+    /// # Examples
+    /// ```no_run
+    /// # use clam::anomaly::automl_regressors::DecisionTreeRegressor;
+    /// #
+    /// // Construct a new DecisionTreeRegressor
+    /// let regressor = DecisionTreeRegressor::new();
+    ///
+    /// // Train and use your regressor
+    /// todo!();
+    /// ```
+    pub fn new() -> Self {
+        Self { model: None }
+    }
+}
+
+impl MetaMLModel for DecisionTreeRegressor {
+    fn train(&mut self, data: MetaMLDataset) {
+        // Construct a metaml dataset from the input data
+        let dataset = metaml_dataset_to_automl(data);
+
+        // Create the settings and model
+        let settings = automl::Settings::default_regression()
+            .only(Algorithm::DecisionTreeRegressor)
+            .with_decision_tree_regressor_settings(DecisionTreeRegressorParameters::default().with_max_depth(3));
```

Comment:
We should make `max_depth` a member of the `DecisionTreeRegressor` struct rather than just hard-coding it here